### ISSUE TITLE
Version 1.9 of dune is not supported

### ISSUE THREAD

**Issue by seantalts** [ocaml/dune#2219]
Hey all,

Not really sure what's going on here, or if this might be user error or something. I have a few scripts that install specific versions of dependencies, including dune 1.8.2 (because there is a bug in dune 1.9 for us) and on one of the machines in CI we're seeing that somehow dune-package files are generated that ask for dune 1.9:

```
#=== ERROR while compiling ppxlib.0.6.0 =======================================#
# context     2.0.4 | macos/x86_64 | ocaml-base-compiler.4.07.1 | https://opam.ocaml.org#b3d552c9
# path        ~/.opam/default/.opam-switch/build/ppxlib.0.6.0
# command     ~/.opam/default/bin/dune build -p ppxlib -j 7
# exit-code   1
# env-file    ~/.opam/log/ppxlib-84016-9ab3aa.env
# output-file ~/.opam/log/ppxlib-84016-9ab3aa.out
### output ###
# File "/Users/Shared/Jenkins/.opam/default/lib/ocaml-compiler-libs/dune-package", line 1, characters 11-14:
# 1 | (lang dune 1.9)
#                ^^^
# Error: Version 1.9 of dune is not supported.
# Supported versions:
# - 0.0
# - 1.0 to 1.8

```

How does this happen? The versions of the software in question are all old enough to predate 1.9 and they don't mention dune in those old versions sometimes, and this is happening across a variety of packages whack-a-mole style. Is there something that is generating dune 1.9 files?

**Comment by seantalts** [comment]
Downgrading to opam 2.0.2 seems to have fixed it!

**Comment by hcarty** [comment]
@seantalts Did you start with a clean `~/.opam` after downgrading to opam 2.0.2?  It looks like you may have been affected by the dune downgrade issues discussed in https://github.com/ocaml/opam/issues/3850

**Comment by seantalts** [comment]
Yep, that looks like the issue. And yeah, I had to completely clean out ~/.opam and downgrade to 2.0.2 to fix it.

**Comment by hcarty** [comment]
For what it's worth, I suspect that the opam downgrade isn't required. Wiping ~/.opam should be sufficient.

**Comment by diml** [comment]
FTR, this will be fixed by #2147. We just need to do a bit more work on the parser
